### PR TITLE
Add retry logic for entrypointer

### PR DESCRIPTION
**Pull request by rick-a-lane-ii**

### CODE REVIEW

File: cmd/relay-runtime-tools/runner.go [hunk 1]
```diff
@@ -49,11 +59,24 @@ func (rr *realRunner) Run(args ...string) error {
...
+
+			return true, nil
+		}, waitOptions...)
+		if err != nil {
 			log.Println(err)
 		}
 
-		if err := validateSchemas(mu); err != nil {
+		err = retry.Wait(ctx, func(ctx context.Context) (bool, error) {
+			if err := validateSchemas(mu); err != nil {
```
Comment (impl) [inline cmd/relay-runtime-tools/runner.go:74]: May be worth noting that once schema validation becomes an actual error we'll have to distinguish between network errors and invalid schema errors here.

**Review by impl — APPROVED** [review]
LGTM, with some minor feedback regarding timeouts in certain circumstances. Feel free to merge when you're happy.

File: cmd/relay-runtime-tools/runner.go [hunk 2]
```diff
@@ -126,21 +173,31 @@ func (rr *realRunner) Run(args ...string) error {
...
+				LogId:   logId,
 				Payload: []byte(line),
 			}
 
 			if ts, err := ptypes.TimestampProto(time.Now().UTC()); err == nil {
 				message.Timestamp = ts
 			}
 
-			postLogMessage(mu, message)
+			err := retry.Wait(ctx, func(ctx context.Context) (bool, error) {
```
[the remark this answers is not in the document]
Comment (rick-a-lane-ii) [reply cmd/relay-runtime-tools/runner.go:190]: Yeah, I started down the line of having customized timeouts for each scenario, but I did not want to add too much additional context handling just yet. I'll need to test a bit first before doing that. We'll see.

I would somewhat prefer we put an overall timeout option (and ideally a hard limit on retry count) as additional options. I appreciate the context-based approach, but it also causes some additional concerns.

File: cmd/relay-runtime-tools/runner.go (hunk 1 above)
Comment (rick-a-lane-ii) [reply cmd/relay-runtime-tools/runner.go:74]: I'm ambivalent about conflating validation with errors, but I might specifically avoid 422s in this code.

**Pull request merged**